media/CircleEditor/view-sidebar.js

```diff
@@ -1456,30 +1438,297 @@ sidebar.ArgumentView = class {
+            'complex64', 'int8', 'float64', 'complex128', 'uint64', 'resource', 'variant', 'uint32'
+        ];
+        if (currentType === 'string') {
+            this.changeBufferType('string', modified, shape);
+        } else {
+            this.compareData(
+                modifiedArr, originalArr, this._editObject._arguments._initializer._data,
+                types.indexOf(currentType.toLowerCase()));
+        }
+    }
+
+    /* buffer type modified - data change NOT detected */
+    changeBufferType(newType, modified, shape) {
+        // original =  textarea.value
+        const modifiedArr = this.validationCheck(modified, shape, newType);
+
+        if (!modifiedArr) {
+            return;
+        }
+
+        const types = [
+            'float32', 'float16', 'int32', 'uint8', 'int64', 'string', 'boolean', 'int16',
+            'complex64', 'int8', 'float64', 'complex128', 'uint64', 'resource', 'variant', 'uint32'
+        ];
 
-    /**
-     * TODO Implement removeBracket()
-     * Bracket remove function
-     */
+        // 0:float, 1:int, 2:uint, 3:string, 4:boolean, 5:complex, 6:resource, 7:variant
+        const typeclass = [0, 0, 1, 2, 1, 3, 4, 1, 5, 1, 0, 5, 2, 6, 7, 2];
+        const bits = [32, 16, 32, 8, 32, 0, 8, 16, 64, 8, 64, 128, 64, 0, 0, 32];
 
-    /**
-     * TODO Implement validationCheck()
-     * Changed buffers validation check function
-     */
+        const buffer = new ArrayBuffer(8);
+        const view = new DataView(buffer);
 
-    /**
-     * TODO Implement editBuffer()
-     * Buffer edit function
-     */
+        const newTypeIndex = types.indexOf(newType.toLowerCase());
 
-    /**
-     * TODO Implement changeBufferType()
-     * Buffer tyep change function
-     */
+        let newArray = [];
+
+        /* string -> ? or ? -> Float or ? -> Int or ? -> Uint */
+        if (typeclass[newTypeIndex] === 0) {  // new type : float
+            for (let i = 0; i < modifiedArr.length; i++) {
+                let data;
+                try {
+                    if (modifiedArr[i] === 'true') {
+                        data = 1;
+                    } else if (modifiedArr[i] === 'false') {
+                        data = 0;
+                    } else {
+                        data = parseFloat(modifiedArr[i]);
+                    }
+                } catch (err) {
+                    return;
+                }
+                if (bits[newTypeIndex] === 16) {  // float16
+                    view.setFloat16(0, data, true);
+                } else if (bits[newTypeIndex] === 32) {  // float32
+                    view.setFloat32(0, data, true);
+                } else if (bits[newTypeIndex] === 64) {  // float64
+                    view.setFloat64(0, data, true);
+                }
+                for (let j = 0; j < bits[newTypeIndex] / 8; j++) {
+                    newArray.push(view.getUint8(j));
+                }
+            }
```

Comment:
With following fixes, `types`, `typeclass`, `bits` variables can be removed.
Furthermore, comments are not needed because variables include the meaning :)

```js
        if (newType.startsWith('float')) {
            const data = parseFloat(modifiedArr[i]);
            const bits = Number(newType.slice(-2));

            for (let i = 0; i < modifiedArr.length; i++) {
                if (bits === 16) {
                    view.setFloat16(0, data, true);
                } else if (bits === 32) {
                    view.setFloat32(0, data, true);
                } else if (bits === 64) {
                    view.setFloat64(0, data, true);
                } else {
                    // Error handling
                }

                for (let j = 0; j < bits / 8; j++) {
                    newArray.push(view.getUint8(j));
                }
            }
```